### ISSUE TITLE
Numblock is not parsed as such in an extension spec as we imply Ruby 2.6 and force to add `:ruby27` metadata

### ISSUE THREAD

**Issue by pirj** [rubocop/rubocop#11479]
Found [in `rubocop-rspec`](https://github.com/rubocop/rubocop-rspec/pull/1564#issuecomment-1399223430).

```
> puts node.source
pending 'not yet implemented' do
    _1
  end

> node
=> s(:block,
  s(:send, nil, :pending,
    s(:str, "not yet implemented")),
  s(:args),
  s(:send, nil, :_1))

> RuboCop::ProcessedSource.new(node.source, RUBY_VERSION.to_f).ast
=> s(:numblock,
  s(:send, nil, :pending,
    s(:str, "not yet implemented")), 1,
  s(:lvar, :_1))
```

--------

## Expected behavior

`node` should be a `numblock`, and `_1` should be an `lvar`, not `send`.

## Actual behavior

`node` is mistakenly parsed as a `block`.

## Steps to reproduce the problem

In [this PR](https://github.com/rubocop/rubocop-rspec/pull/1564), add the following spec:
```ruby
# spec/rubocop/cop/rspec/pending_block_inside_example_spec.rb
    it 'registers an offense when using `pending` with a numblock' do
      expect_offense(<<~RUBY)
        it "does something" do
          pending 'not yet implemented' do
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Don't pass a block to `pending` or `skip` inside examples.
            _1
          end
        end
      RUBY
    end
```

Set a debug breakpoint:
```
# lib/rubocop/cop/rspec/pending_block_inside_example.rb:43
        def on_block(node) # rubocop:disable InternalAffairs/NumblockHandler
          require 'pry-byebug'; binding.pry
```

## RuboCop version

```
$ rubocop -V
```
1.43.0 (using Parser 3.2.0.0, rubocop-ast 1.24.1, running on ruby 3.0.3) [x86_64-darwin18]
  - rubocop-performance 1.15.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.18.1
```

**Comment by pirj** [comment]
I've debugged this down to:
```
From: /Users/pirj/.rvm/gems/ruby-3.0.3@rubocop-rspec/gems/rubocop-1.43.0/lib/rubocop/rspec/cop_helper.rb:24:
Owner: CopHelper
Visibility: public
Signature: parse_source(source, file=?)
Number of lines: 12

def parse_source(source, file = nil)
  if file.respond_to?(:write)
    file.write(source)
    file.rewind
    file = file.path
  end

  processed_source = RuboCop::ProcessedSource.new(source, ruby_version, file)
  processed_source.config = configuration
  processed_source.registry = registry
  processed_source
end
[17] pry(#<RSpec::ExampleGroups::RuboCopCopRSpecPendingBlockInsideExample::WhenUsingInsideExample>)> ruby_version
=> 2.6
[18] pry(#<RSpec::ExampleGroups::RuboCopCopRSpecPendingBlockInsideExample::WhenUsingInsideExample>)> RUBY_VERSION
=> "3.0.3"
```

Which comes from [cop_helper.rb](https://github.com/rubocop/rubocop/blob/3a542a96fb59c254d68348c54e04af8fdcab7b4e/lib/rubocop/rspec/cop_helper.rb#L9):
```ruby
let(:ruby_version) { 2.6 }
```

Should it be 2.6 by default?
Should we remember to use `:ruby27` metadata when writing specs for cops that support newer syntax?
```ruby
  context 'Ruby 2.7', :ruby27 do
    it 'autocorrects array.sort_by { |x| x }' do
      expect_offense(<<~RUBY)
        array.sort_by { _1 }
              ^^^^^^^^^^^^^^ Use `sort` instead of `sort_by { _1 }`.
      RUBY
```

What would be the downside of using
```ruby
let(:ruby_version) { RUBY_VERSION.to_f }
```
by default?

**Comment by koic** [comment]
Yes. Ruby 2.7's numbered block parameter syntax is an incompatible AST between Ruby 2.6 and Ruby 2.7. This is because `_1` can be used as a variable or method name prior Ruby 2.6.

```console
% ruby-parse --26 -e 'foo { _1 }'
(block
  (send nil :foo)
  (args)
  (send nil :_1))

% ruby-parse --27 -e 'foo { _1 }'
(numblock
  (send nil :foo) 1
  (lvar :_1))
```

And the current default `:ruby_version` value is 2.6. Please specify `:ruby27` metadata when using Ruby 2.7's numbered block parameter.

**Comment by pirj** [comment]
Understood, thank you.

I've played around with changing the parser from 2.6 to 2.7 by default in specs, but faced some specs that I have no intention of changing.

I'll close this as it's a common practice to test numblock code with adding metadata, even though it can be missed.

```
     Failure/Error:
       raise 'Error parsing example code: ' \
             "#{processed_source.diagnostics.map(&:render).join("\n")}"

     RuntimeError:
       Error parsing example code: (string):1:27: error: circular argument reference some_arg
       (string):1: def some_method(some_arg: some_arg)
       (string):1:                           ^~~~~~~~
       (string):2:8: error: circular argument reference some_arg
       (string):2:   puts some_arg
       (string):2:        ^~~~~~~~
     # ./lib/rubocop/rspec/expect_offense.rb:203:in `parse_processed_source'
     # ./lib/rubocop/rspec/expect_offense.rb:119:in `expect_offense'
     # ./spec/rubocop/cop/lint/circular_argument_reference_spec.rb:60:in `block (4 levels) in <top (required)>'
     # /home/pirj/.rvm/gems/ruby-3.0.1@rubocop/gems/webmock-3.18.1/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'

  4) RuboCop::Cop::Style::RescueModifier registers an offense for modifier rescue around parallel assignment
     Failure/Error: expect(actual_annotations).to eq(expected_annotations), ''

       Diff:
       @@ -1,3 +1,3 @@
        a, b = 1, 2 rescue nil
       -^^^^^^^^^^^^^^^^^^^^^^ Avoid using `rescue` in its modifier form.
       +       ^^^^^^^^^^^^^^^ Avoid using `rescue` in its modifier form.

     # ./lib/rubocop/rspec/expect_offense.rb:123:in `expect_offense'
     # ./spec/rubocop/cop/style/rescue_modifier_spec.rb:22:in `block (2 levels) in <top (required)>'
     # /home/pirj/.rvm/gems/ruby-3.0.1@rubocop/gems/webmock-3.18.1/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'

  5) RuboCop::Cop::Style::ParallelAssignment corrects when the expression uses a modifier rescue statement
     Failure/Error: expect(actual_annotations).to eq(expected_annotations), ''

       Diff:
       @@ -1,3 +1,2 @@
        a, b = 1, 2 rescue foo
       -^^^^^^^^^^^ Do not use parallel assignment.
```
